Makefile

```diff
@@ -12,50 +12,93 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-BINS = $(wildcard velero-*)
+# The binary to build (just the basename).
+BIN ?= $(wildcard velero-*)
 
-REPO ?= github.com/heptio/velero-plugin-example
+# This repo's root import path (under GOPATH).
+PKG := github.com/heptio/velero-plugin-example
 
-BUILD_IMAGE ?= gcr.io/heptio-images/golang:1.9-alpine3.6
+BUILD_IMAGE ?= golang:1.11-alpine3.8
 
 IMAGE ?= gcr.io/heptio-images/velero-plugin-example
 
-ARCH ?= amd64
+# Which architecture to build - see $(ALL_ARCH) for options.
+# if the 'local' rule is being run, detect the ARCH from 'go env'
+# if it wasn't specified by the caller.
+local : ARCH ?= $(shell go env GOOS)-$(shell go env GOARCH)
+ARCH ?= linux-amd64
 
-all: $(addprefix build-, $(BINS))
+platform_temp = $(subst -, ,$(ARCH))
+GOOS = $(word 1, $(platform_temp))
+GOARCH = $(word 2, $(platform_temp))
+
+# TODO(ncdc): support multiple image architectures once gcr.io supports manifest lists
+# Set default base image dynamically for each arch
+ifeq ($(GOARCH),amd64)
```

Comment:
This section doesn't look like it's used (or valid for this repo), please remove